api.go

```diff
@@ -987,3 +988,228 @@ func (app *App) APIGetChallengeSkin() func(c echo.Context) error {
 		})
 	})
 }
+
+type APITokenResponse struct {
+	Token string `json:"token" example:"dfghjdrtjdrfr"`
+}
+
+type APILoginRequest struct {
+	Username string `json:"username" example:"notch"`
+	Password string `json:"password" example:"OIXtjQeASfz"`
+}
+
+// APILogin godoc
+//
+//	@Summary	  	Get a token
+//	@Description	Get a token for login credentials.
+//	@Tags		  		users, auth
+//	@Accept				json
+//	@Produce			json
+//	@Success			200	{object}	APITokenResponse
+//	@Failure			400	{object}	APIError
+//	@Failure			401	{object}	APIError
+//	@Failure			423	{object}	APIError
+//	@Failure			500	{object}	APIError
+//	@Router				/drasl/api/v2/login [post]
+func (app *App) APILogin() func(c echo.Context) error {
```

Comment:
These new routes need to be rate-limited (see `makeRateLimiter` in main.go). Then the HTTP 429 failure code needs to be documented here in the API doc.